src/appmixer/slack/list/ListUsers/ListUsers.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const commons = require('../../slack-commons');
+
+const outputPortName = 'out';
+const TTL_USERS = 20 * 1000; // 20 sec
+
+module.exports = {
+
+    /**
+     * @link https://api.slack.com/methods/users.list
+     */
+    async receive(context) {
+
+        const generateOutputPortOptions = context.properties.generateOutputPortOptions;
+        let { outputType, limit, isSource } = context.messages.in.content;
+
+        if (generateOutputPortOptions) {
+            return this.getOutputPortOptions(context, outputType);
+        }
+
+        const cacheKey = 'slack-list-users-' + context.flowId;
+        let lock;
+        try {
+            lock = await context.lock(context.flowId, { retryDelay: 500 });
+
+            if (isSource) {
+                const usersCached = await context.staticCache.get(cacheKey);
+                if (usersCached) {
+                    await commons.sendArrayOutput({ context, outputType, records: usersCached });
+                    return;
+                }
+            }
+
+            const { data } = await context.httpRequest({
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${context.auth.accessToken}`
+                },
+                url: 'https://slack.com/api/users.list',
+                params: { limit }
+            });
+
+            if (!data) {
+                throw new context.CancelError(response.statusText);
```

Comment:
is this correct? won't we lose the re-try queue on network errors?

Reply:
`isSource` should ignore all errors and returns an empty list when error occurs, correct?

Reply:
I now changed it to throw `Error` so that it can automatically retry. The original code was inspired by `makeRequest` from `slack-commons.js` which is used for all other Slack API calls. This may need refactoring as well.